### PR TITLE
Various Improvements and Refactors to Test Suite

### LINKED ISSUE #6: Find a way to test the initialization of the registry's domain separator
See https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/a5b3e4db42c7230a4ae1eab0e8bca6c723c0ea1d#r1471961987 for context

### PR DESCRIPTION
**Pull request by apbendi**
Hey @garyghayrat, rather than cause too much churn on your PR, I did some cleanup and refactoring to your PR in my own branch. You can review it, leave feedback, and once you're ok with my changes, we can merge it in to your branch. Then we can ask Matt & Tony for final sign off on your PR before merging it into main and getting community input in advance of the audit. Thanks for bearing with me in this process!

Also note, the only non-test change was to update the DOMAIN_SEPARATOR to correctly confirm to the spec. The name of the param should always be `verifyingContract`.

resolves #6

### CODE REVIEW

File: test/ERC6538Registry.t.sol [hunk 1]
```diff
@@ -343,168 +320,58 @@ contract RegisterKeysOnBehalf is ERC6538RegistryTest {
...
-    pure
-    returns (address, RecoverError, bytes32)
+  function testFuzz_ReturnsARecomputedValueIfTheChainIdChanges(uint256 chainId1, uint256 chainId2)
+    external
   {
-    if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0) {
-      return (address(0), RecoverError.InvalidSignatureS, s);
-    }
+    chainId1 = bound(chainId1, 1, 2 ** 64 - 1);
+    chainId2 = bound(chainId2, 1, 2 ** 64 - 1);
```
Comment (garyghayrat) [inline test/ERC6538Registry.t.sol:354]: why do we wanna use `2 ** 64 - 1` here?

**Review by garyghayrat — APPROVED** [review]
I see how this greatly simplifies the test suite, and I learned a couple tricks along the way. Curious about a number that you chose, otherwise looks good to me. Thank you!

File: test/ERC6538Registry.t.sol [hunk 2]
```diff
@@ -79,260 +132,184 @@ contract RegisterKeys is ERC6538RegistryTest {
...
-      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
-    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
-    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
-    bytes memory signature = abi.encodePacked(r, s, v);
+    (address notRegistrant, uint256 notRegistrantPrivateKey) = makeAddrAndKey(seed);
+    vm.assume(notRegistrant != registrant);
+    bytes memory signature =
+      _generateRegistrationSignature(notRegistrantPrivateKey, schemeId, stealthMetaAddress, 0);
 
     vm.expectRevert();
```
Comment (garyghayrat) [inline test/ERC6538Registry.t.sol]: Just got a comment from matt we might want to implement wherever we have `vm.expectRevert()`. Comment:https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5#discussion_r1476213212

Comment (garyghayrat) [reply test/ERC6538Registry.t.sol]: Updated here [fc2a267](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/7/commits/fc2a267daa35427360d802230678342f94ba3bbc)

File: test/ERC6538Registry.t.sol (hunk 1 above)
Comment (apbendi) [reply test/ERC6538Registry.t.sol:354]: This is the highest supported chainId so there's no reason not to fuzz across the whole range.

**Pull request merged**